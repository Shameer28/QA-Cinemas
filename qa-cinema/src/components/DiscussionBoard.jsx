import {useState} from "react"

import Tab from 'react-bootstrap/Tab'
import Tabs from 'react-bootstrap/Tabs'
import DiscussionRatings from "./props/DiscussionRatings";

const DiscussionBoard = () => {
	const [key, setKey] = useState("reviews");

	return (  
		<div>
			<h3>
				Welcome to the discussion board
			</h3>
			<Tabs
				id="controlled-tab-example"
				activeKey={key}
				onSelect={(k) => setKey(k)}
			>
				<Tab eventKey="reviews" title="Ratings">
					<DiscussionRatings/>	
				</Tab>
				<Tab eventKey="forum" title="Film Discussion">
					Lorem ipsum dolor sit amet consectetur adipisicing elit. Odit cumque quisquam necessitatibus eligendi excepturi illo eveniet, architecto temporibus. Quasi, aliquid. Blanditiis, facere debitis. Temporibus inventore quis ad nulla sint sed!
				</Tab>
			</Tabs>
		</div>
	)
}

export default DiscussionBoard;