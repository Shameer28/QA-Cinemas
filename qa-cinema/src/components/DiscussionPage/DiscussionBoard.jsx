import { useState } from "react"

import Tab from 'react-bootstrap/Tab'
import Tabs from 'react-bootstrap/Tabs'
import DiscussionRatings from "./DiscussionRatings";
import Forum from "./Forum";
import { Button, Image, Container } from 'react-bootstrap';
import image1 from './../img/review.jpg';

const DiscussionBoard = () => {
	const [key, setKey] = useState("reviews");

	return (
		<div className="background">
			<Image src={image1} alt="review banner" width="100% fluid" /> <br /><br />
			<Container >
				<h1 style={{ color: "white", fontWeight: "400", fontfamily: "isonormregular, sans-serif", letterSpacing: ".15em", textTransform: "uppercase", lineHeight: "1.1" }}>
					Welcome to the Discussion Board
				</h1>
				<Tabs
					id="controlled-tab-example"
					activeKey={key}
					onSelect={(k) => setKey(k)}
				>
					<Tab eventKey="reviews" title="Ratings">
						<DiscussionRatings />
					</Tab>
					<Tab eventKey="forum" title="Film Discussion">
						<Forum />
					</Tab>
				</Tabs>
			</Container>
		</div >
	)
}

export default DiscussionBoard;