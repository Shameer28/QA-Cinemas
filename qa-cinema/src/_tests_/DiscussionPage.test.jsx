import DiscussionBoard from "../components/DiscussionPage/DiscussionBoard";
import { create } from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";


describe(`Discussion board testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<BrowserRouter><DiscussionBoard /></BrowserRouter>).toJSON();
        expect(instance).toMatchSnapshot();
    })
})
//Click on one of tabs to fully test- COME BACK TO THIS!