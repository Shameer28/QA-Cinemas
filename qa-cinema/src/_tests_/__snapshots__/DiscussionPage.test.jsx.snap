// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discussion board testing Should match the snapshot 1`] = `
<div
  className="background"
>
  <img
    alt="review banner"
    className=""
    src="review.jpg"
    width="100% fluid"
  />
   
  <br />
  <br />
  <div
    className="container"
  >
    <h1
      style={
        Object {
          "color": "white",
          "fontWeight": "400",
          "fontfamily": "isonormregular, sans-serif",
          "letterSpacing": ".15em",
          "lineHeight": "1.1",
          "textTransform": "uppercase",
        }
      }
    >
      Welcome to the Discussion Board
    </h1>
    <nav
      className="nav nav-tabs"
      onKeyDown={[Function]}
      role="tablist"
    >
      <a
        aria-controls="controlled-tab-example-tabpane-reviews"
        aria-selected={true}
        className="nav-item nav-link active"
        data-rb-event-key="reviews"
        href="#"
        id="controlled-tab-example-tab-reviews"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
      >
        Ratings
      </a>
      <a
        aria-controls="controlled-tab-example-tabpane-forum"
        aria-selected={false}
        className="nav-item nav-link"
        data-rb-event-key="forum"
        href="#"
        id="controlled-tab-example-tab-forum"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
      >
        Film Discussion
      </a>
    </nav>
    <div
      className="tab-content"
    >
      <div
        aria-hidden={false}
        aria-labelledby="controlled-tab-example-tab-reviews"
        className="fade tab-pane active show"
        id="controlled-tab-example-tabpane-reviews"
        role="tabpanel"
      >
        <div>
          <a
            href="/reviews"
            onClick={[Function]}
          >
            <br />
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
            >
              Add a Review
            </button>
          </a>
          <br />
          <br />
          <h3
            style={
              Object {
                "color": "white",
                "fontWeight": "400",
                "fontfamily": "isonormregular, sans-serif",
                "letterSpacing": ".15em",
                "lineHeight": "1.1",
                "textTransform": "uppercase",
              }
            }
          >
            Recent Reviews
          </h3>
          <br />
          <div>
            <br />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="controlled-tab-example-tab-forum"
        className="fade tab-pane"
        id="controlled-tab-example-tabpane-forum"
        role="tabpanel"
      >
        <div>
          <br />
          <div>
            <div />
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Add Thread
            </button>
          </div>
           
          <br />
          <br />
          <h3
            style={
              Object {
                "color": "white",
                "fontWeight": "400",
                "fontfamily": "isonormregular, sans-serif",
                "letterSpacing": ".15em",
                "lineHeight": "1.1",
                "textTransform": "uppercase",
              }
            }
          >
            Forum
          </h3>
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
